### PR TITLE
[master] Merge DNNL adaptive pooling with standard pooling

### PR DESCRIPTION
**Pull request by piotrwolinski-intel**
## Description ##

DNNL adaptive and standard pooling now have one implementation with single type of Parameter  - `PoolingParam`

### CODE REVIEW

File: src/operator/contrib/adaptive_avg_pooling.cc [hunk 1]
```diff
@@ -199,7 +202,53 @@ void AdaptiveAvgPoolUpdateGradInput(mshadow::Stream<cpu>* s,
...
+
+void AdaptiveAvgPoolComputeExCPU(const nnvm::NodeAttrs& attrs,
+                                 const OpContext& ctx,
+                                 const std::vector<NDArray>& inputs,
+                                 const std::vector<OpReqType>& req,
+                                 const std::vector<NDArray>& outputs) {
+  CHECK_EQ(inputs.size(), 1U);
+  CHECK_EQ(outputs.size(), 1U);
+  /*
+  OneDNN doesn't support adaptive pooling.
```
Comment (bartekkuncer) [inline src/operator/contrib/adaptive_avg_pooling.cc:238]: oneDNN*
Also is this comment up-to-date?

Comment (piotrwolinski-intel) [reply src/operator/contrib/adaptive_avg_pooling.cc]: Yes, from what I've checked it is still up-to-date, but some verification would be appreciated

**Comment by piotrwolinski-intel** [comment]
@mxnet-bot run ci [unix-cpu]

**Comment by piotrwolinski-intel** [comment]
@mxnet-bot run ci [unix-cpu]

**Comment by piotrwolinski-intel** [comment]
@mxnet-bot run ci [unix-cpu]

**Review by bartekkuncer — APPROVED** [review]

File: src/operator/nn/pooling.cc [hunk 2]
```diff
@@ -290,7 +290,7 @@ void PoolingComputeExCPU(const nnvm::NodeAttrs& attrs,
       workspace = &outputs[1];
     }
     DNNL_OPCHECK_INIT(false, 1, inputs, outputs);
-    DNNLPoolingCompute(ctx, param, inputs[0], req[0], outputs[0], workspace);
+    DNNLPoolingCompute(ctx, param, inputs[0], req[0], outputs[0], workspace, false);
```
Comment (bgawrych) [inline src/operator/nn/pooling.cc:293]: I wonder if we can run it like all others ops.
DNNL_OPCHECK_INIT...
DNNLRun(DNNLPoolingCompute...
DNNL_OPCHECK_RUN...

This one additional bool parameter can be passed by template argument and IsView check in checking support function won't be necessary. What do you think? + @bartekkuncer

**Review by bgawrych — APPROVED** [review]
LGTM

File: src/operator/nn/pooling.cc (hunk 2 above)
Comment (bartekkuncer) [reply src/operator/nn/pooling.cc:293]: Making it consistent with other operators is desirable. @piotrwolinski-intel Please check if it introduces any problems.

Template here might make the code more readable as well, especially with `true/false` boolean parameter changed to simple `enum class PoolingType { regular, adaptive };`

Comment (bartekkuncer) [reply src/operator/nn/pooling.cc:293]: In my opinion as @piotrwolinski-intel is working currently on backward implementation for this op, this PR could be merged as is and the changes we are talking about can be applied along with backward propagation addition. Do you agree @bgawrych ?

Comment (bgawrych) [reply src/operator/nn/pooling.cc:293]: Ok, I'm merging it now and looking forward for backward refactor

**Pull request merged**